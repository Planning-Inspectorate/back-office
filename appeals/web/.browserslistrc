# https://browserslist.dev/?q=PiAxJSwgbGFzdCAyIHZlcnNpb25zLCBub3QgaWUgPD0gMTE%3D
> 1%
last 2 versions
Chrome > 105
Edge > 105,
not ie <= 11
