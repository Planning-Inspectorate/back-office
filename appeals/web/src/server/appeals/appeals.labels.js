export const APPEAL_TYPE = 'Appeal type';
