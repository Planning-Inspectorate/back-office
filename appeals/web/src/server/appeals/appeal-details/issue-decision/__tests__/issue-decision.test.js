import { parseHtml } from '@pins/platform';
import nock from 'nock';
import supertest from 'supertest';
import { createTestEnvironment } from '#testing/index.js';
import { mapDecisionOutcome } from '../issue-decision.mapper.js';
import { documentFileInfo, inspectorDecisionData } from '#testing/appeals/appeals.js';

const { app, installMockApi, teardown } = createTestEnvironment();
const request = supertest(app);
const baseUrl = '/appeals-service/appeal-details';
const issueDecisionPath = '/issue-decision';
const decisionPath = '/decision';
const decisionLetterUploadPath = '/decision-letter-upload';
const decisionLetterDatePath = '/decision-letter-date';
const checkYourDecisionPath = '/check-your-decision';

describe('issue-decision', () => {
	beforeEach(installMockApi);
	afterEach(teardown);

	describe('GET /change-appeal-type/appeal-type', () => {
		it('should render the decision page', async () => {
			const response = await request.get(`${baseUrl}/1${issueDecisionPath}/${decisionPath}`);
			const element = parseHtml(response.text);

			expect(element.innerHTML).toMatchSnapshot();
		});
	});
	describe('POST /:appealId/issue-decision/decision', () => {
		beforeEach(() => {
			nock('http://test/').get('/appeals/1').reply(200, inspectorDecisionData);
			nock('http://test/').get('/appeals/1/documents/1').reply(200, documentFileInfo);
		});
		afterEach(teardown);

		it('should render the decision details correctly', async () => {
			const mockAppealId = '1';
			const mockAppealDecision = 'allowed';
			const response = await request
				.post(`${baseUrl}/${mockAppealId}/issue-decision/decision`)
				.send({ decision: mockAppealDecision })
				.expect(302);

			expect(response.headers.location).toBe(
				`/appeals-service/appeal-details/${mockAppealId}/issue-decision/decision-letter-upload`
			);
		});

		it('should send the decision details for invalid, and redirect correctly', async () => {
			const mockAppealId = '1';
			const mockAppealDecision = 'Invalid';
			const response = await request
				.post(`${baseUrl}/${mockAppealId}/issue-decision/decision`)
				.send({ decision: mockAppealDecision })
				.expect(302);

			expect(response.headers.location).toBe(
				`/appeals-service/appeal-details/${mockAppealId}/issue-decision/invalid-reason`
			);
		});
	});

	describe('GET /change-appeal-type/decision-letter-upload', () => {
		it('should render the decision letter upload page', async () => {
			const response = await request.get(
				`${baseUrl}/1${issueDecisionPath}/${decisionLetterUploadPath}`
			);
			const element = parseHtml(response.text);

			expect(element.innerHTML).toMatchSnapshot();
		});
	});

	describe('GET /change-appeal-type/decision-letter-date', () => {
		it('should render the decision letter date page', async () => {
			const response = await request.get(
				`${baseUrl}/1${issueDecisionPath}/${decisionLetterDatePath}`
			);
			const element = parseHtml(response.text);

			expect(element.innerHTML).toMatchSnapshot();
		});
	});

	describe('POST /:appealId/issue-decision/decision-letter-date', () => {
		beforeEach(() => {
			nock('http://test/').get('/appeals/1').reply(200, inspectorDecisionData);
			nock('http://test/').get('/appeals/1/documents/1').reply(200, documentFileInfo);
		});
		afterEach(teardown);

		it('should render the decision letter date correctly', async () => {
			const mockAppealId = '1';
			const mockLetterDecisionDate = {
				'decision-letter-date-day': '1',
				'decision-letter-date-month': '1',
				'decision-letter-date-year': '2023'
			};
			const response = await request
				.post(`${baseUrl}/${mockAppealId}/issue-decision/decision-letter-date`)
				.send(mockLetterDecisionDate)
				.expect(302);

			expect(response.headers.location).toBe(
				`/appeals-service/appeal-details/${mockAppealId}/issue-decision/check-your-decision`
			);
		});

		it('should re-render the  decision letter date page with an error message if the provided date day is invalid', async () => {
			const response = await request
				.post(`${baseUrl}/1${issueDecisionPath}/${decisionLetterDatePath}`)
				.send({
					'decision-letter-date-day': 0,
					'decision-letter-date-month': 11,
					'decision-letter-date-year': 2024
				});

			expect(response.statusCode).toBe(200);
			const element = parseHtml(response.text);
			expect(element.innerHTML).toMatchSnapshot();
		});

		it('should re-render the  decision letter date page with an error message if the provided date month is invalid', async () => {
			const response = await request
				.post(`${baseUrl}/1${issueDecisionPath}/${decisionLetterDatePath}`)
				.send({
					'decision-letter-date-day': 1,
					'decision-letter-date-month': 0,
					'decision-letter-date-year': 2024
				});

			expect(response.statusCode).toBe(200);
			const element = parseHtml(response.text);
			expect(element.innerHTML).toMatchSnapshot();
		});

		it('should re-render the  decision letter date page with an error message if the provided date year is invalid', async () => {
			const response = await request
				.post(`${baseUrl}/1${issueDecisionPath}/${decisionLetterDatePath}`)
				.send({
					'decision-letter-date-day': 1,
					'decision-letter-date-month': 11,
					'decision-letter-date-year': 'x'
				});

			expect(response.statusCode).toBe(200);
			const element = parseHtml(response.text);
			expect(element.innerHTML).toMatchSnapshot();
		});

		it('should re-render the  decision letter date page with an error message if the provided date year is not in the past', async () => {
			teardown;
			const response = await request
				.post(`${baseUrl}/1${issueDecisionPath}/${decisionLetterDatePath}`)
				.send({
					'decision-letter-date-day': 1,
					'decision-letter-date-month': 11,
					'decision-letter-date-year': 3000
				});

			expect(response.statusCode).toBe(200);
			const element = parseHtml(response.text);
			expect(element.innerHTML).toMatchSnapshot();
		});

		it('should re-render the  decision letter date page with an error message if the provided date is not a business day', async () => {
			teardown;
			const response = await request
				.post(`${baseUrl}/1${issueDecisionPath}/${decisionLetterDatePath}`)
				.send({
					'decision-letter-date-day': 25,
					'decision-letter-date-month': 12,
					'decision-letter-date-year': 2024
				});

			expect(response.statusCode).toBe(200);
			const element = parseHtml(response.text);
			expect(element.innerHTML).toMatchSnapshot();
		});
	});

	describe('GET /invalid-reason', () => {
		it('should render the invalid reason page', async () => {
			const mockAppealId = '1';
			const response = await request.get(
				`${baseUrl}/${mockAppealId}/issue-decision/invalid-reason`
			);

			const element = parseHtml(response.text);

			expect(element.innerHTML).toMatchSnapshot();
		});
	});

	describe('POST /invalid-reason', () => {
		it('should process the invalid reason page and redirect', async () => {
			const mockAppealId = '1';
			const mockReason = 'Reasons!';
			const response = await request
				.post(`${baseUrl}/${mockAppealId}/issue-decision/invalid-reason`)
				.send({ decisionInvalidReason: mockReason })
				.expect(302);

			expect(response.headers.location).toBe(
				`/appeals-service/appeal-details/${mockAppealId}/issue-decision/check-invalid-decision`
			);
		});
	});

	describe('GET /change-appeal-type/check-your-decision', () => {
		it('should render the check your decision letter page', async () => {
			const response = await request.get(
				`${baseUrl}/1${issueDecisionPath}/${checkYourDecisionPath}`
			);
			const element = parseHtml(response.text);

			expect(element.innerHTML).toMatchSnapshot();
		});
	});

	describe('GET /check-invalid-decision', () => {
		it('should render the check your answer page for invalid decision', async () => {
			const mockAppealId = '1';
			const mockReason = 'Reasons!';
			const invalidReasonResponse = await request
				.post(`${baseUrl}/${mockAppealId}/issue-decision/invalid-reason`)
				.send({ decisionInvalidReason: mockReason })
				.expect(302);

			expect(invalidReasonResponse.headers.location).toBe(
				`/appeals-service/appeal-details/${mockAppealId}/issue-decision/check-invalid-decision`
			);

			const response = await request.get(`${baseUrl}/1${issueDecisionPath}/check-invalid-decision`);
			const element = parseHtml(response.text);

			expect(element.innerHTML).toMatchSnapshot();
		});
	});
});

describe('mapDecisionOutcome', () => {
	it('should map "allowed" to "Allowed"', () => {
		const result = mapDecisionOutcome('allowed');
		expect(result).toBe('Allowed');
	});

	it('should map "dismissed" to "Dismissed"', () => {
		const result = mapDecisionOutcome('dismissed');
		expect(result).toBe('Dismissed');
	});

	it('should map "split" to "Split decision"', () => {
		const result = mapDecisionOutcome('split');
		expect(result).toBe('Split decision');
	});

	it('should return an empty string for undefined outcome', () => {
		const result = mapDecisionOutcome(undefined);
		expect(result).toBe('');
	});

	it('should return an empty string for invalid outcome', () => {
		const result = mapDecisionOutcome('invalid');
		expect(result).toBe('');
	});
});
