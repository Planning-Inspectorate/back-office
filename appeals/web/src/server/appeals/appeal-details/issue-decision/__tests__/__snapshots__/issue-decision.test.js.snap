// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`issue-decision GET /change-appeal-type/appeal-type should render the decision page 1`] = `
"<main class="govuk-main-wrapper " id="main-content" role="main">
    <div class="govuk-grid-row">
        <div class="govuk-grid-column-two-thirds">
            <div class="govuk-grid-row">
                <div class="govuk-grid-column-full"><span class="govuk-caption-l govuk-!-margin-bottom-3">Appeal 351062</span>
                    <h1                     class="govuk-heading-l govuk-!-margin-bottom-3">What is the decision?</h1>
                </div>
            </div>
            <form method="POST" novalidate="novalidate">
                <div class="govuk-form-group">
                    <fieldset class="govuk-fieldset">
                        <legend class="govuk-fieldset__legend govuk-fieldset__legend--m">Issue your decision</legend>
                        <div class="govuk-radios" data-module="govuk-radios">
                            <div class="govuk-radios__item">
                                <input class="govuk-radios__input" id="decision" name="decision" type="radio"
                                value="Allowed">
                                <label class="govuk-label govuk-radios__label" for="decision">Allowed</label>
                            </div>
                            <div class="govuk-radios__item">
                                <input class="govuk-radios__input" id="decision-2" name="decision" type="radio"
                                value="Dismissed">
                                <label class="govuk-label govuk-radios__label" for="decision-2">Dismissed</label>
                            </div>
                            <div class="govuk-radios__item">
                                <input class="govuk-radios__input" id="decision-3" name="decision" type="radio"
                                value="Split">
                                <label class="govuk-label govuk-radios__label" for="decision-3">Split Decision</label>
                            </div>
                        </div>
                    </fieldset>
                </div>
                <button type="submit" class="govuk-button" data-module="govuk-button">Continue</button>
            </form>
        </div>
    </div>
</main>"
`;

exports[`issue-decision GET /change-appeal-type/check-your-decision should render the check your decision letter page 1`] = `
"<main class="govuk-main-wrapper " id="main-content" role="main">
    <div class="govuk-grid-row">
        <div class="govuk-grid-column-two-thirds">
            <div class="govuk-grid-row">
                <div class="govuk-grid-column-full"><span class="govuk-caption-l govuk-!-margin-bottom-3">Appeal 351062</span>
                    <h1                     class="govuk-heading-l govuk-!-margin-bottom-3">Check your answers</h1>
                </div>
            </div>
            <form method="POST" novalidate="novalidate">
                <dl class="govuk-summary-list">
                    <div class="govuk-summary-list__row"><dt class="govuk-summary-list__key"> Decision</dt>
                        <dd class="govuk-summary-list__value"></dd>
                        <dd class="govuk-summary-list__actions"><a class="govuk-link" href="/appeals-service/appeal-details/1/issue-decision/decision"> Change</a>
                        </dd>
                    </div>
                    <div class="govuk-summary-list__row"><dt class="govuk-summary-list__key"> Decision letter</dt>
                        <dd class="govuk-summary-list__value"></dd>
                        <dd class="govuk-summary-list__actions"><a class="govuk-link" href="/appeals-service/appeal-details/1/issue-decision/decision-letter-upload"> Change</a>
                        </dd>
                    </div>
                    <div class="govuk-summary-list__row"><dt class="govuk-summary-list__key"> Decision date</dt>
                        <dd class="govuk-summary-list__value"></dd>
                        <dd class="govuk-summary-list__actions"><a class="govuk-link" href="/appeals-service/appeal-details/1/issue-decision/decision-letter-date"> Change</a>
                        </dd>
                    </div>
                </dl>
                <div class="govuk-warning-text"><span class="govuk-warning-text__icon" aria-hidden="true">!</span>
                    <strong                     class="govuk-warning-text__text"><span class="govuk-warning-text__assistive">Warning</span> You are about
                        to send the decision to all parties and close the appeal. Please make sure
                        you have reviewed the decision information</strong>
                </div>
                <div class="govuk-checkboxes__item govuk-!-margin-bottom-5">
                    <input class="govuk-checkboxes__input" id="ready-to-send" name="ready-to-send"
                    type="checkbox" value="Yes">
                    <label class="govuk-label govuk-checkboxes__label" for="ready-to-send">This decision is ready to be sent to all parties</label>
                </div>
                <button type="submit" class="govuk-button" data-module="govuk-button">Submit this decision</button>
            </form>
        </div>
    </div>
</main>"
`;

exports[`issue-decision GET /change-appeal-type/decision-letter-date should render the decision letter date page 1`] = `
"<main class="govuk-main-wrapper " id="main-content" role="main">
    <div class="govuk-grid-row">
        <div class="govuk-grid-column-two-thirds">
            <div class="govuk-grid-row">
                <div class="govuk-grid-column-full"><span class="govuk-caption-l govuk-!-margin-bottom-3">Appeal 351062</span>
                    <h1                     class="govuk-heading-l govuk-!-margin-bottom-3">Tell us the date on your decision letter</h1>
                </div>
            </div>
            <form method="POST" novalidate="novalidate">
                <div class="govuk-form-group">
                    <fieldset class="govuk-fieldset" role="group" aria-describedby="decision-letter-date-hint">
                        <div id="decision-letter-date-hint" class="govuk-hint">For example, 27 3 2023</div>
                        <div class="govuk-date-input" id="decision-letter-date">
                            <div class="govuk-date-input__item">
                                <div class="govuk-form-group">
                                    <label class="govuk-label govuk-date-input__label" for="decision-letter-date-day">Day</label>
                                    <input class="govuk-input govuk-date-input__input govuk-input govuk-date-input__input govuk-input--width-2"
                                    id="decision-letter-date-day" name="decision-letter-date-day" type="text"
                                    inputmode="numeric">
                                </div>
                            </div>
                            <div class="govuk-date-input__item">
                                <div class="govuk-form-group">
                                    <label class="govuk-label govuk-date-input__label" for="decision-letter-date-month">Month</label>
                                    <input class="govuk-input govuk-date-input__input govuk-input govuk-date-input__input govuk-input--width-2"
                                    id="decision-letter-date-month" name="decision-letter-date-month" type="text"
                                    inputmode="numeric">
                                </div>
                            </div>
                            <div class="govuk-date-input__item">
                                <div class="govuk-form-group">
                                    <label class="govuk-label govuk-date-input__label" for="decision-letter-date-year">Year</label>
                                    <input class="govuk-input govuk-date-input__input govuk-input govuk-date-input__input govuk-input--width-4"
                                    id="decision-letter-date-year" name="decision-letter-date-year" type="text"
                                    inputmode="numeric">
                                </div>
                            </div>
                        </div>
                    </fieldset>
                </div>
                <button type="submit" class="govuk-button" data-module="govuk-button">Continue</button>
            </form>
        </div>
    </div>
</main>"
`;

exports[`issue-decision GET /change-appeal-type/decision-letter-upload should render the decision letter upload page 1`] = `
"<main class="govuk-main-wrapper " id="main-content" role="main">
    <div class="govuk-grid-row">
        <div class="govuk-grid-column-full"><span class="govuk-caption-l govuk-!-margin-bottom-3">Appeal 351062</span>
            <h1             class="govuk-heading-l govuk-!-margin-bottom-3">Upload your decision letter</h1>
        </div>
    </div>
    <div class="govuk-warning-text"><span class="govuk-warning-text__icon" aria-hidden="true">!</span>
        <strong         class="govuk-warning-text__text"><span class="govuk-warning-text__assistive">Warning</span> Before uploading
            the decision letter check that you:</strong>
    </div>
    <ul class="govuk-list govuk-list--bullet">
        <li>have used the correct appeal reference</li>
        <li>have added the decision and visit dates</li>
        <li>have added the correct site address</li>
        <li>have added the decision to the top and bottom of the letter</li>
        <li>signed the letter</li>
    </ul>
    <div class="govuk-grid-row pins-file-upload" data-next-page-url="/appeals-service/appeal-details/1/issue-decision/decision-letter-date"
    data-case-id="1" data-folder-id="123" data-document-type="decisionLetter"
    data-document-stage="appeal_decision" data-blob-storage-host="https://127.0.0.1:10000/devstoreaccount1"
    data-blob-storage-container="document-service-uploads" data-allowed-types=".pdf,application/pdf,.doc,application/msword,.docx,application/vnd.openxmlformats-officedocument.wordprocessingml.document,.ppt,application/vnd.ms-powerpoint,.pptx,application/vnd.openxmlformats-officedocument.presentationml.presentation,.xls,application/vnd.ms-excel,.xlsx,application/vnd.openxmlformats-officedocument.spreadsheetml.sheet,.jpg,image/jpeg,.jpeg,image/jpeg,.mpeg,video/mpeg,.mp3,audio/mpeg,.mp4,video/mp4,.mov,video/quicktime,.png,image/png,.tif,image/tiff,.tiff,image/tiff,">
        <div class="top-errors-hook"></div>
        <h1 class="govuk-heading-m govuk-grid-column-full">Upload decision letter</h1>
        <form class="govuk-grid-column-two-thirds">
            <div class="pins-file-upload--container">
                <div class="govuk-body colour--secondary pins-file-upload--instructions">
                    <label for="upload-file-1">Choose a single file to upload.</label><span>Your file must be PDF, DOC, DOCX, PPT, PPTX, XLS, XLSX, JPG, JPEG, MPEG, MP3, MP4, MOV, PNG, TIF, 	or TIFF.</span><span>The total of your uploaded files must be smaller than 1 GB.</span>
                    <div                     class="middle-errors-hook"></div>
            </div>
            <div class="display--flex">
                <input class="display--none" id="upload-file-1" accept=".pdf,application/pdf,.doc,application/msword,.docx,application/vnd.openxmlformats-officedocument.wordprocessingml.document,.ppt,application/vnd.ms-powerpoint,.pptx,application/vnd.openxmlformats-officedocument.presentationml.presentation,.xls,application/vnd.ms-excel,.xlsx,application/vnd.openxmlformats-officedocument.spreadsheetml.sheet,.jpg,image/jpeg,.jpeg,image/jpeg,.mpeg,video/mpeg,.mp3,audio/mpeg,.mp4,video/mp4,.mov,video/quicktime,.png,image/png,.tif,image/tiff,.tiff,image/tiff,"
                type="file" name="files" value="Choose file" aria-controls="file-list-1">
                <button type='button' class="pins-file-upload--button govuk-button">Choose file</button><span role="status" aria-live="assertive" class='govuk-body pins-file-upload--counter'> No file chosen</span>
            </div>
            <div id="file-list-1">
                <h2 class="display--sr-only" id="file-list-title-1">List of files to upload</h2>
                <ul class="pins-file-upload--files-rows" aria-describedby="file-list-title-1"
                aria-live="polite"></ul>
            </div>
    </div>
    <div class="govuk-inset-text">Saving files will default them to ‘Unredacted’ and ‘Not checked’</div>
    <div     class="display--flex">
        <button class="govuk-button pins-file-upload--submit" data-module="govuk-button"
        id="submit-button-1">Continue</button>
        <div class="progress-hook"></div>
        </div>
        </form>
        </div>
</main>"
`;

exports[`issue-decision POST /:appealId/issue-decision/decision-letter-date should re-render the  decision letter date page with an error message if the provided date day is invalid 1`] = `
"<main class="govuk-main-wrapper " id="main-content" role="main">
    <div class="govuk-grid-row">
        <div class="govuk-grid-column-two-thirds">
            <div class="govuk-error-summary" data-module="govuk-error-summary">
                <div role="alert">
                    <h2 class="govuk-error-summary__title"> There is a problem</h2>
                    <div class="govuk-error-summary__body">
                        <ul class="govuk-list govuk-error-summary__list">
                            <li><a href="#decision-letter-date-day">Decision letter date day must be between 1 and 31</a>
                            </li>
                            <li><a href="#">Decision letter date must be a valid date</a>
                            </li>
                        </ul>
                    </div>
                </div>
            </div>
            <div class="govuk-grid-row">
                <div class="govuk-grid-column-full"><span class="govuk-caption-l govuk-!-margin-bottom-3">Appeal 351062</span>
                    <h1                     class="govuk-heading-l govuk-!-margin-bottom-3">Tell us the date on your decision letter</h1>
                </div>
            </div>
            <form method="POST" novalidate="novalidate">
                <div class="govuk-form-group">
                    <fieldset class="govuk-fieldset" role="group" aria-describedby="decision-letter-date-hint">
                        <div id="decision-letter-date-hint" class="govuk-hint">For example, 27 3 2023</div>
                        <div class="govuk-date-input" id="decision-letter-date">
                            <div class="govuk-date-input__item">
                                <div class="govuk-form-group">
                                    <label class="govuk-label govuk-date-input__label" for="decision-letter-date-day">Day</label>
                                    <input class="govuk-input govuk-date-input__input govuk-input govuk-date-input__input govuk-input--width-2"
                                    id="decision-letter-date-day" name="decision-letter-date-day" type="text"
                                    value="0" inputmode="numeric">
                                </div>
                            </div>
                            <div class="govuk-date-input__item">
                                <div class="govuk-form-group">
                                    <label class="govuk-label govuk-date-input__label" for="decision-letter-date-month">Month</label>
                                    <input class="govuk-input govuk-date-input__input govuk-input govuk-date-input__input govuk-input--width-2"
                                    id="decision-letter-date-month" name="decision-letter-date-month" type="text"
                                    value="11" inputmode="numeric">
                                </div>
                            </div>
                            <div class="govuk-date-input__item">
                                <div class="govuk-form-group">
                                    <label class="govuk-label govuk-date-input__label" for="decision-letter-date-year">Year</label>
                                    <input class="govuk-input govuk-date-input__input govuk-input govuk-date-input__input govuk-input--width-4"
                                    id="decision-letter-date-year" name="decision-letter-date-year" type="text"
                                    value="2024" inputmode="numeric">
                                </div>
                            </div>
                        </div>
                    </fieldset>
                </div>
                <button type="submit" class="govuk-button" data-module="govuk-button">Continue</button>
            </form>
        </div>
    </div>
</main>"
`;

exports[`issue-decision POST /:appealId/issue-decision/decision-letter-date should re-render the  decision letter date page with an error message if the provided date month is invalid 1`] = `
"<main class="govuk-main-wrapper " id="main-content" role="main">
    <div class="govuk-grid-row">
        <div class="govuk-grid-column-two-thirds">
            <div class="govuk-error-summary" data-module="govuk-error-summary">
                <div role="alert">
                    <h2 class="govuk-error-summary__title"> There is a problem</h2>
                    <div class="govuk-error-summary__body">
                        <ul class="govuk-list govuk-error-summary__list">
                            <li><a href="#decision-letter-date-month">Decision letter date month must be between 1 and 12</a>
                            </li>
                            <li><a href="#">Decision letter date must be a valid date</a>
                            </li>
                        </ul>
                    </div>
                </div>
            </div>
            <div class="govuk-grid-row">
                <div class="govuk-grid-column-full"><span class="govuk-caption-l govuk-!-margin-bottom-3">Appeal 351062</span>
                    <h1                     class="govuk-heading-l govuk-!-margin-bottom-3">Tell us the date on your decision letter</h1>
                </div>
            </div>
            <form method="POST" novalidate="novalidate">
                <div class="govuk-form-group">
                    <fieldset class="govuk-fieldset" role="group" aria-describedby="decision-letter-date-hint">
                        <div id="decision-letter-date-hint" class="govuk-hint">For example, 27 3 2023</div>
                        <div class="govuk-date-input" id="decision-letter-date">
                            <div class="govuk-date-input__item">
                                <div class="govuk-form-group">
                                    <label class="govuk-label govuk-date-input__label" for="decision-letter-date-day">Day</label>
                                    <input class="govuk-input govuk-date-input__input govuk-input govuk-date-input__input govuk-input--width-2"
                                    id="decision-letter-date-day" name="decision-letter-date-day" type="text"
                                    value="1" inputmode="numeric">
                                </div>
                            </div>
                            <div class="govuk-date-input__item">
                                <div class="govuk-form-group">
                                    <label class="govuk-label govuk-date-input__label" for="decision-letter-date-month">Month</label>
                                    <input class="govuk-input govuk-date-input__input govuk-input govuk-date-input__input govuk-input--width-2"
                                    id="decision-letter-date-month" name="decision-letter-date-month" type="text"
                                    value="0" inputmode="numeric">
                                </div>
                            </div>
                            <div class="govuk-date-input__item">
                                <div class="govuk-form-group">
                                    <label class="govuk-label govuk-date-input__label" for="decision-letter-date-year">Year</label>
                                    <input class="govuk-input govuk-date-input__input govuk-input govuk-date-input__input govuk-input--width-4"
                                    id="decision-letter-date-year" name="decision-letter-date-year" type="text"
                                    value="2024" inputmode="numeric">
                                </div>
                            </div>
                        </div>
                    </fieldset>
                </div>
                <button type="submit" class="govuk-button" data-module="govuk-button">Continue</button>
            </form>
        </div>
    </div>
</main>"
`;

exports[`issue-decision POST /:appealId/issue-decision/decision-letter-date should re-render the  decision letter date page with an error message if the provided date year is invalid 1`] = `
"<main class="govuk-main-wrapper " id="main-content" role="main">
    <div class="govuk-grid-row">
        <div class="govuk-grid-column-two-thirds">
            <div class="govuk-error-summary" data-module="govuk-error-summary">
                <div role="alert">
                    <h2 class="govuk-error-summary__title"> There is a problem</h2>
                    <div class="govuk-error-summary__body">
                        <ul class="govuk-list govuk-error-summary__list">
                            <li><a href="#decision-letter-date-year">Decision letter date year must be a number</a>
                            </li>
                            <li><a href="#">Decision letter date must be a valid date</a>
                            </li>
                        </ul>
                    </div>
                </div>
            </div>
            <div class="govuk-grid-row">
                <div class="govuk-grid-column-full"><span class="govuk-caption-l govuk-!-margin-bottom-3">Appeal 351062</span>
                    <h1                     class="govuk-heading-l govuk-!-margin-bottom-3">Tell us the date on your decision letter</h1>
                </div>
            </div>
            <form method="POST" novalidate="novalidate">
                <div class="govuk-form-group">
                    <fieldset class="govuk-fieldset" role="group" aria-describedby="decision-letter-date-hint">
                        <div id="decision-letter-date-hint" class="govuk-hint">For example, 27 3 2023</div>
                        <div class="govuk-date-input" id="decision-letter-date">
                            <div class="govuk-date-input__item">
                                <div class="govuk-form-group">
                                    <label class="govuk-label govuk-date-input__label" for="decision-letter-date-day">Day</label>
                                    <input class="govuk-input govuk-date-input__input govuk-input govuk-date-input__input govuk-input--width-2"
                                    id="decision-letter-date-day" name="decision-letter-date-day" type="text"
                                    value="1" inputmode="numeric">
                                </div>
                            </div>
                            <div class="govuk-date-input__item">
                                <div class="govuk-form-group">
                                    <label class="govuk-label govuk-date-input__label" for="decision-letter-date-month">Month</label>
                                    <input class="govuk-input govuk-date-input__input govuk-input govuk-date-input__input govuk-input--width-2"
                                    id="decision-letter-date-month" name="decision-letter-date-month" type="text"
                                    value="11" inputmode="numeric">
                                </div>
                            </div>
                            <div class="govuk-date-input__item">
                                <div class="govuk-form-group">
                                    <label class="govuk-label govuk-date-input__label" for="decision-letter-date-year">Year</label>
                                    <input class="govuk-input govuk-date-input__input govuk-input govuk-date-input__input govuk-input--width-4"
                                    id="decision-letter-date-year" name="decision-letter-date-year" type="text"
                                    value="x" inputmode="numeric">
                                </div>
                            </div>
                        </div>
                    </fieldset>
                </div>
                <button type="submit" class="govuk-button" data-module="govuk-button">Continue</button>
            </form>
        </div>
    </div>
</main>"
`;

exports[`issue-decision should re-render the  decision letter date page with an error message if the provided date is not a business day 1`] = `
"<main class="govuk-main-wrapper " id="main-content" role="main">
    <div class="govuk-grid-row">
        <div class="govuk-grid-column-two-thirds">
            <div class="govuk-error-summary" data-module="govuk-error-summary">
                <div role="alert">
                    <h2 class="govuk-error-summary__title"> There is a problem</h2>
                    <div class="govuk-error-summary__body">
                        <ul class="govuk-list govuk-error-summary__list">
                            <li><a href="#">Date must be today or in the past</a>
                            </li>
                        </ul>
                    </div>
                </div>
            </div>
            <div class="govuk-grid-row">
                <div class="govuk-grid-column-full"><span class="govuk-caption-l govuk-!-margin-bottom-3">Appeal 351062</span>
                    <h1                     class="govuk-heading-l govuk-!-margin-bottom-3">Tell us the date on your decision letter</h1>
                </div>
            </div>
            <form method="POST" novalidate="novalidate">
                <div class="govuk-form-group">
                    <fieldset class="govuk-fieldset" role="group" aria-describedby="decision-letter-date-hint">
                        <div id="decision-letter-date-hint" class="govuk-hint">For example, 27 3 2023</div>
                        <div class="govuk-date-input" id="decision-letter-date">
                            <div class="govuk-date-input__item">
                                <div class="govuk-form-group">
                                    <label class="govuk-label govuk-date-input__label" for="decision-letter-date-day">Day</label>
                                    <input class="govuk-input govuk-date-input__input govuk-input govuk-date-input__input govuk-input--width-2"
                                    id="decision-letter-date-day" name="decision-letter-date-day" type="text"
                                    value="25" inputmode="numeric">
                                </div>
                            </div>
                            <div class="govuk-date-input__item">
                                <div class="govuk-form-group">
                                    <label class="govuk-label govuk-date-input__label" for="decision-letter-date-month">Month</label>
                                    <input class="govuk-input govuk-date-input__input govuk-input govuk-date-input__input govuk-input--width-2"
                                    id="decision-letter-date-month" name="decision-letter-date-month" type="text"
                                    value="12" inputmode="numeric">
                                </div>
                            </div>
                            <div class="govuk-date-input__item">
                                <div class="govuk-form-group">
                                    <label class="govuk-label govuk-date-input__label" for="decision-letter-date-year">Year</label>
                                    <input class="govuk-input govuk-date-input__input govuk-input govuk-date-input__input govuk-input--width-4"
                                    id="decision-letter-date-year" name="decision-letter-date-year" type="text"
                                    value="2024" inputmode="numeric">
                                </div>
                            </div>
                        </div>
                    </fieldset>
                </div>
                <button type="submit" class="govuk-button" data-module="govuk-button">Continue</button>
            </form>
        </div>
    </div>
</main>"
`;

exports[`issue-decision should re-render the  decision letter date page with an error message if the provided date year is not in the past 1`] = `
"<main class="govuk-main-wrapper " id="main-content" role="main">
    <div class="govuk-grid-row">
        <div class="govuk-grid-column-two-thirds">
            <div class="govuk-error-summary" data-module="govuk-error-summary">
                <div role="alert">
                    <h2 class="govuk-error-summary__title"> There is a problem</h2>
                    <div class="govuk-error-summary__body">
                        <ul class="govuk-list govuk-error-summary__list">
                            <li><a href="#">Date must be today or in the past</a>
                            </li>
                        </ul>
                    </div>
                </div>
            </div>
            <div class="govuk-grid-row">
                <div class="govuk-grid-column-full"><span class="govuk-caption-l govuk-!-margin-bottom-3">Appeal 351062</span>
                    <h1                     class="govuk-heading-l govuk-!-margin-bottom-3">Tell us the date on your decision letter</h1>
                </div>
            </div>
            <form method="POST" novalidate="novalidate">
                <div class="govuk-form-group">
                    <fieldset class="govuk-fieldset" role="group" aria-describedby="decision-letter-date-hint">
                        <div id="decision-letter-date-hint" class="govuk-hint">For example, 27 3 2023</div>
                        <div class="govuk-date-input" id="decision-letter-date">
                            <div class="govuk-date-input__item">
                                <div class="govuk-form-group">
                                    <label class="govuk-label govuk-date-input__label" for="decision-letter-date-day">Day</label>
                                    <input class="govuk-input govuk-date-input__input govuk-input govuk-date-input__input govuk-input--width-2"
                                    id="decision-letter-date-day" name="decision-letter-date-day" type="text"
                                    value="1" inputmode="numeric">
                                </div>
                            </div>
                            <div class="govuk-date-input__item">
                                <div class="govuk-form-group">
                                    <label class="govuk-label govuk-date-input__label" for="decision-letter-date-month">Month</label>
                                    <input class="govuk-input govuk-date-input__input govuk-input govuk-date-input__input govuk-input--width-2"
                                    id="decision-letter-date-month" name="decision-letter-date-month" type="text"
                                    value="11" inputmode="numeric">
                                </div>
                            </div>
                            <div class="govuk-date-input__item">
                                <div class="govuk-form-group">
                                    <label class="govuk-label govuk-date-input__label" for="decision-letter-date-year">Year</label>
                                    <input class="govuk-input govuk-date-input__input govuk-input govuk-date-input__input govuk-input--width-4"
                                    id="decision-letter-date-year" name="decision-letter-date-year" type="text"
                                    value="3000" inputmode="numeric">
                                </div>
                            </div>
                        </div>
                    </fieldset>
                </div>
                <button type="submit" class="govuk-button" data-module="govuk-button">Continue</button>
            </form>
        </div>
    </div>
</main>"
`;
